src/gui/layertree/qgslayertreeview.cpp

```diff
@@ -253,20 +253,22 @@ void QgsLayerTreeView::updateExpandedStateFromNode( QgsLayerTreeNode *node )
 
 QgsMapLayer *QgsLayerTreeView::layerForIndex( const QModelIndex &index ) const
 {
-  QgsLayerTreeNode *node = layerTreeModel()->index2node( index );
-  if ( node )
+  if ( index.isValid( ) )
   {
```

Comment:
These changes prevent a segfault that I run into when developing the test, I'm not sure that it will be happen in the real world use case, but better stay on the safe side.

The reason was that the model was not set, in that case the index was also invalid, In fact I was thinking to also check the model in addition to (or instead of) the check on the index validity:
```
if ( layerTreeModel() && index.isValid( ) )
{
...
}
```